src/components/common/LanguageSwitch.tsx

```diff
@@ -0,0 +1,38 @@
+"use client"
+
+import { getAvailableLanguages } from "~/lib/i18n/client"
+
+import { Menu } from "../ui/Menu"
+
+export function LanguageSwitch() {
+  const languages = getAvailableLanguages()
+
+  const changeLanguage = (language: string) => {
+    document.cookie = `preferred_language=${language};${document.cookie}`
```

Comment:
Every time the language is switched, `preferred_language` field will be repeatedly added.

Reply:
From what I've seen, it doesn't seem to be, it looks like a direct replacement. The following is the log when i change the language.

![ScreenShot 2023-08-16 10 12 43](https://github.com/Crossbell-Box/xLog/assets/38493346/17362509-a6c3-4c26-862f-3567ff40d4be)